File: Problem_set_1/Introductory_problems/p03_Repetition/Repetitions.cpp
#include<iostream>
#include<string>
#include<algorithm>

using namespace std;

int main(){
    string input;

    getline(cin, input);

    int max_streak = 1, current_streak = 1;

    for(int i = 1; i < input.length(); i++){
        if(input[i] == input[i - 1]){
            current_streak++;
        } else{
            max_streak = max(max_streak, current_streak);
            current_streak = 1;
        }
    }

    if(current_streak > max_streak){
        max_streak = current_streak;
    }

    cout << max_streak;

    return 0;
}

File: Problem_set_1/Introductory_problems/p03_Repetition/Repetitions.ts
const input = require("prompt-sync")();

const line = String(input());

let max_streak: number = 1;
let current_streak: number = 1;

for(let i: number = 1; i < line.length; i++){
    if(line[i] === line[i - 1]){
        current_streak += 1;
    } else{
        max_streak = Math.max(max_streak, current_streak);
        current_streak = 1;
    }
}

if(current_streak > max_streak){
    max_streak = current_streak;
}

console.log(max_streak);